sycl/plugins/esimd_emulator/pi_esimd_emulator.cpp

```diff
@@ -1063,35 +1086,35 @@ pi_result piMemRelease(pi_mem Mem) {
   }
 
   if (--(Mem->RefCount) == 0) {
-    if (Mem->getMemType() == PI_MEM_TYPE_BUFFER) {
-      _pi_buffer *PiBuf = static_cast<_pi_buffer *>(Mem);
-      // TODO implement libCM API failure logging mechanism, so that these
-      // failures are clearly distinguishable from other EMU plugin failures.
-      int Result =
-          Mem->Context->Device->CmDevicePtr->DestroySurface(PiBuf->CmBufferPtr);
-
-      if (Result != cm_support::CM_SUCCESS) {
+    if (Mem->isCmSurface()) {
```

Comment:
As we talked, we need to review the design of handling scenarios where host buffer memory is created by SYCL runtime, then we can proceed with this PR.

Reply:
Updated after design review.